auction-server/src/auction.rs

```diff
@@ -1066,3 +1034,353 @@ async fn simulate_bid_svm(chain_store: &ChainStoreSvm, bid: &BidSvm) -> Result<(
         None => Ok(()),
     }
 }
+
+pub trait Auction {
+    type Block: DebugTrait;
+    type BlockStream<'a>: Stream<Item = Self::Block> + Unpin + Send + 'a;
+    type WsClient;
+    type SimulatedBid: SimulatedBidTrait + Clone;
+    type ConclusionResult;
+
+    const CHAIN_TYPE: models::ChainType;
+    const AUCTION_MINIMUM_LIFETIME: Duration;
+
+    fn get_ws_client(self) -> impl Future<Output = Result<Self::WsClient>> + Send;
+    fn get_block_stream<'a>(
+        client: &'a Self::WsClient,
+    ) -> impl Future<Output = Result<Self::BlockStream<'a>>>;
+    fn filter_bids(bids: Vec<SimulatedBid>) -> Vec<Self::SimulatedBid>;
+    fn get_winner_bids(
```

Comment:
I think now that we have these trait functions, it's worth documenting what each of these functions are supposed to do